samples/net/coap_client/prj.conf

```diff
@@ -16,6 +16,15 @@ CONFIG_NET_CONNECTION_MANAGER=y
 
 # CoAP
 CONFIG_COAP=y
+CONFIG_COAP_CLIENT=y
```

Comment:
Gathering all the `COAP_CLIENT` related configurations would make it look a bit nicer.